translib/transformer/xfmr_intf.go

```diff
@@ -2452,17 +2562,177 @@ var Subscribe_intf_ip_addr_xfmr = func(inParams XfmrSubscInParams) (XfmrSubscOut
 	if log.V(3) {
 		log.Info("Entering Subscribe_intf_ip_addr_xfmr")
 	}
+	var err error
 	var result XfmrSubscOutParams
 
 	pathInfo := NewPathInfo(inParams.uri)
 	origTargetUriPath := pathInfo.YangPath
 
 	log.Infof("Subscribe_intf_ip_addr_xfmr:- subscProc:%v URI: %s", inParams.subscProc, inParams.uri)
 	log.Infof("Subscribe_intf_ip_addr_xfmr:- Target URI path: %s", origTargetUriPath)
-	// Defer the DB resource check done by infra by setting the virtual table to true.
-	// Resource checks are now performed within the DbToYang or YangToDb subtree callback.
-	result.isVirtualTbl = true
-	return result, nil
+
+	// When the subscribe subtree is invoked in the GET or CRUD context the inParams.subscProc is set to TRANSLATE_EXISTS
+	if inParams.subscProc == TRANSLATE_EXISTS {
+		// Defer the DB resource check done by infra by setting the virtual table to true.
+		// Resource checks are now performed within the DbToYang or YangToDb subtree callback.
+		result.isVirtualTbl = true
+		return result, nil
+	}
+	if inParams.subscProc == TRANSLATE_SUBSCRIBE {
+
+		ifBasePath := "/openconfig-interfaces:interfaces/interface"
+		targetUriPath := origTargetUriPath[len(ifBasePath):]
+
+		if strings.HasPrefix(targetUriPath, "/subinterfaces") {
+			targetUriPath = targetUriPath[len("/subinterfaces/subinterface"):]
+		}
+		if strings.HasPrefix(targetUriPath, "/openconfig-if-ip:ipv4") {
+			targetUriPath = targetUriPath[len("/openconfig-if-ip:ipv4/addresses"):]
+		} else {
+			targetUriPath = targetUriPath[len("/openconfig-if-ip:ipv6/addresses"):]
+		}
+
+		if targetUriPath == "" || targetUriPath == "/address" {
+			result.isVirtualTbl = true
+			log.Info("Subscribe_intf_ip_addr_xfmr:- result.isVirtualTbl: ", result.isVirtualTbl)
+			return result, err
+		}
+
+		result.onChange = OnchangeEnable
+		result.nOpts = &notificationOpts{}
+		result.nOpts.pType = OnChange
+		result.isVirtualTbl = false
+
+		uriIfName := pathInfo.Var("name")
+		tableName := ""
+		ipKey := ""
+		ifKey := ""
+
+		if uriIfName == "" || uriIfName == "*" {
+			ifKey = "*"
+		} else {
+			sonicIfName := &uriIfName
+			ifKey = *sonicIfName
```

Comment:
Why take an address and then dereference? I think just the following is sufficient -- `ifKey := pathInfo.StringVar("name", "*")`